### PR TITLE
fix: Evict txs from mempool exceeding size limits

### PR DESCRIPTION
**Pull request by davidterpay**

### CODE REVIEW

**Review by aljo242 — APPROVED** [review]

File: lanes/mev/abci.go [hunk 1]
```diff
@@ -49,7 +49,6 @@ func (h *ProposalHandler) PrepareLaneHandler() base.PrepareLaneHandler {
 			}
 
 			cacheCtx, write := ctx.CacheContext()
-
```
Comment (nivasan1) [inline lanes/mev/abci.go:52]: Why remove?

**Review by nivasan1 — APPROVED** [review]
lgtm

**Review by Eric-Warehime — COMMENTED** [review]
A test for this would be nice, and pretty easy to write if you just set max gas/bytes to small numbers right?

**Comment by davidterpay** [comment]
> A test for this would be nice, and pretty easy to write if you just set max gas/bytes to small numbers right?
There are tests for this, just added an additional condition check to ensure the tx was removed from the mempool

**Comment by Eric-Warehime** [comment]
Screenshot 2024-06-25 at 8 22 24 AM
cc @davidterpay

**Comment by davidterpay** [comment]
not sure why codecov is saying no coverage when the test case hits that exact line of code. check `lanes/base/abci_test.go`

cc: @Eric-Warehime

**Comment by davidterpay** [comment]
Screenshot 2024-06-25 at 11 25 48 AM

**Pull request merged**